crates/mpc-protocols/src/mpc_service.rs

```diff
@@ -0,0 +1,117 @@
+use crate::mpc_events::{CreatedProofMPCEvent, MPCEvent};
+use log::info;
+use pera_types::base_types::ObjectID;
+use pera_types::event::Event;
+use std::cell::RefCell;
+use std::collections::{HashMap, VecDeque};
+use tokio::sync::{mpsc, Mutex, RwLock};
+
+// TODO (#253): Make this configurable from a config file
+const MAX_ACTIVE_MPC_INSTANCES: usize = 100;
+
+pub enum MPCInput {
+    InitEvent(CreatedProofMPCEvent),
+    Message,
+}
+
+struct MPCInstance {
+    status: MPCSessionStatus,
+    /// The channel to send message to this instance's message handler thread when this instance is active
+    input_receiver: Option<mpsc::Sender<MPCInput>>,
+    pending_messages: Vec<MPCInput>,
+}
+
+/// Possible status of an MPC session
+/// - Active: The session is currently running, new messages will be forwarded to the session
+/// - Pending: The session is waiting for a slot to become active, when the init received there were already `MAX_ACTIVE_MPC_INSTANCES` active sessions
+/// - Finished: The session has finished, no more messages will be forwarded. The session will be removed from the service after a timeout.
+/// We want to keep it for some time so leftover messages related to it won't be treated as malicious.
```

Comment:
- what is a "slot"? please be more specific about this
- "when the init received there were already `MAX_ACTIVE_MPC_INSTANCES` active sessions". This is business logic, which is prone to change. Drop it and let the interface speak about that.
- "The session will be removed from the service after a timeout." again, implementation. Perhaps compact that a bit with the first sentence.
- "/// We want to keep it for some time so leftover messages related to it won't be treated as malicious." again, implementation. This comment should be closer to the logic taking care of awaiting removal.

```suggestion
/// Possible statuses of an MPC session:
/// - Active: The session is currently running; new messages will be forwarded to the session.
/// - Pending: The session is awaiting a slot <WHAT IS A SLOT?> to become active.
/// - Finished: The session is finished and pending removal; incoming messages will not be forwarded.
```

Reply:
Please do not accept the above change immediately but update it to deal with the question.